lib/shelley/src/Cardano/Wallet/Shelley/Launch.hs

```diff
@@ -362,13 +424,22 @@ newtype PoolConfig = PoolConfig
     }
     deriving (Eq, Show)
 
+data RunningNode = RunningNode
+    FilePath
+    -- ^ Socket path
+    Block
+    -- ^ Genesis block
+    (NetworkParameters, NodeVersionData)
+
 -- | Execute an action after starting a cluster of stake pools. The cluster also
 -- contains a single BFT node that is pre-configured with keys available in the
 -- test data.
 --
 -- This BFT node is essential in order to bootstrap the chain and allow
 -- registering pools. Passing `0` as a number of pool will simply start a single
 -- BFT node.
+--
+-- The callback actions are not guaranteed to use the same node.
```

Comment:
This comment looks scary. It's quite important that we use the relay node for sending transactions, otherwise they might be rolled back and screw the entire test setup.

Reply:
The existing pool setup are using the BFT node to setup the stake pools.

As the PR is now, the BFT would also be used to transfer funds to the shelley faucet addresses.

Are you sure that is a problem? I think the code is looks slightly nicer if we don't have to start the relay until the end, as we used to.

Reply:
It shouldn't be a problem indeed since there won't be any stake pools at the time the faucet funds are transferred to cause any rollback.